packages/core/metrics/core-metrics-server-internal/src/routes/load_window.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import assert from 'node:assert';
+
+/** We .ceil to rather _slightly_ over-report usage in certain circumstances */
+const twoDeci = (num: number) => Math.ceil(num * 100) / 100;
+
+export class LoadWindow {
+  readonly #window: number[];
+
+  constructor(private readonly size: number) {
+    this.#window = new Array(size).fill(0);
+  }
+
+  addObservation(value: number) {
+    this.#window.unshift(Math.max(0, value));
+    this.#window.pop();
+    assert(
+      this.#window.length === this.size,
+      `Expected window size of ${this.size}, but received ${this.#window.length}`
+    );
```

Comment:
Thinking about performance... do we need to run this assertion on every tick?

Also thinking about the collateral damage... what would happen to the collector if it fails? Will Kibana crash?

Reply:
> Thinking about performance... do we need to run this assertion on every tick?

This should be a relatively cheap check :)

> Also thinking about the collateral damage...

This is based on "offensive programming" they use [in Chromium](https://www.chromium.org/chromium-os/developer-library/reference/cpp/cpp-invariant-checks/#check-dcheck-notreached_noreturn-and-notimplemented). The idea is that if ever this invariant is violated the consequences are worse than the process crashing. For this code I figured correctness is mission critical for autoscaling, but perhaps you're write that it's overkill -- I'll remove it. We can rely on tests to check this invariant.